src/lib/block_producer/block_producer.ml

```diff
@@ -959,7 +959,9 @@ let run ~context:(module Context : CONTEXT) ~vrf_evaluator ~prover ~verifier
                         [%log info] ~metadata
                           "Generated transition $state_hash was accepted into \
                            transition frontier" ;
-                        return ()
+                        Deferred.map ~f:Result.return
```

Comment:
Shouldn't this be a `don't_wait_for`? We don't actually care when the broadcast happens in this context.